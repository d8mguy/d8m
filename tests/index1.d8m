
// index1
// This test is to trigger the predicate index rwrule with dict-like search. It should subst -1 for nil
val prT = tuple(ky : string, vl : label)
val aLst = [[prT: "ekrj", :xx], [prT: "iuio", :yx], [prT: "xxxx", :xy], [prT: "xxxxy", :yy], [prT: "3", :x3]]
val foo = \(lst : list(prT)) {
    val inx = lst[this.ky == "3" => index]
    inx == nil ? "boo" : "at #{inx} got #{lst[inx].vl}"
}
foo(aLst)
