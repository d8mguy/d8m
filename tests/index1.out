at 4 got x3
